services/identityagentsrv/service.go

```diff
@@ -1,98 +1,170 @@
 package identityagentsrv
 
 import (
-	"bytes"
-	"encoding/hex"
-	"errors"
+	// "bytes"
+	// "encoding/hex"
+	// "errors"
+	"fmt"
 
+	"github.com/dghubble/sling"
+	common3 "github.com/iden3/go-iden3-core/common"
 	"github.com/iden3/go-iden3-core/core"
 	"github.com/iden3/go-iden3-core/db"
 	"github.com/iden3/go-iden3-core/merkletree"
+	"github.com/iden3/go-iden3-core/services/claimsrv"
+	"gopkg.in/go-playground/validator.v9"
 )
 
 // identityagentsrv is a service that can hold multiple IdentityAgents (one for each Identity)
 
-var PREFIX_EMITTEDCLAIMS = []byte("emittedclaims")
-var PREFIX_RECEIVEDCLAIMS = []byte("receivedclaims")
-
-type Service interface {
-	LoadIdStorages(id *core.ID) (*IdStorages, error)
-	NewIdentity(claimAuthKOp merkletree.Claim, extraGenesisClaims []merkletree.Claim) (*core.ID, *core.ProofClaim, error)
-	AddClaim(id *core.ID, claim merkletree.Claim) error
-	AddClaims(id *core.ID, claims []merkletree.Claim) error
-	GetAllReceivedClaims(id *core.ID, idStorages *IdStorages) ([]core.ClaimObj, error)
-	GetAllEmittedClaims(id *core.ID, idStorages *IdStorages) ([]core.ClaimObj, error)
-	GetAllClaims(id *core.ID) ([]core.ClaimObj, []core.ClaimObj, error)
-	GetFullMT(id *core.ID) (map[string]string, error)
+var PREFIX_CLAIMSEMITTED = []byte("claimsemitted")
+var PREFIX_CLAIMSRECEIVED = []byte("claimsreceived")
+var PREFIX_CLAIMSGENESIS = []byte("claimsgenesis")
+
+// TODO: Move this to a generic place
+type ServerError struct {
+	Err string `json:"error"`
 }
 
-type RootUpdaterConfig struct {
-	Type   string            // "relay", "ethereum"
-	Params map[string]string // if type=="relay" -> "url"
+// TODO: Move this to a generic place
+func (e ServerError) Error() string {
+	return fmt.Sprintf("server: %v", e.Err)
 }
 
-type ServiceImpl struct {
-	storage     db.Storage
-	rootUpdater RootUpdaterConfig
+type RootUpdaterMock struct{}
+
+func (ru *RootUpdaterMock) RootUpdate(id *core.ID, setRootReq claimsrv.SetRoot0Req) error {
+	return fmt.Errorf("mock mock")
 }
 
-func New(storage db.Storage) *ServiceImpl {
-	return &ServiceImpl{
-		storage: storage,
+func (ru *RootUpdaterMock) GetRootProof(id *core.ID) (*core.ProofClaim, error) {
+	return nil, fmt.Errorf("mock mock")
+}
+
+func (ru *RootUpdaterMock) ClaimAuthService() *merkletree.Entry { return nil }
+
+type RootUpdaterRelay struct {
+	RelayUrl string
+	RelayId  *core.ID
+	_client  *sling.Sling
```

Comment:
why `_client`? I'm agree on Rust implementation of iden3, but maybe not yet ^^ :crab: